### PR TITLE
[exporter/datadog] Add `host_metadata` section

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:**

Add `host_metadata` section to group related settings.

**Link to tracking Issue:** #9099

**Testing:** 

- [x] Added unit tests
- [x] Check that deprecation warnings are logged
- [x] Test that host metadata keeps being sent when `only_metadata` is false (not covered by tests)

**Documentation:** Updated documentation

### CODE REVIEW

File: exporter/datadogexporter/config/config.go
Comment (albertvaka) [inline exporter/datadogexporter/config/config.go:34]:
```suggestion
	errNoMetadata  = errors.New("only_metadata can't be enabled when host_metadata::enabled = false or host_metadata::hostname_source != first_resource")
```

**Review by albertvaka — APPROVED** [review]

**Comment by jpkrohling** [comment]
I trust @mx-psi, but would it be possible to get an approval from another code owner as well? Like @KSerrania, @gbbr or @knusbaum

**Comment by mx-psi** [comment]
@jpkrohling Sure, I will ask internally for this

**Review by KSerrania — APPROVED** [review]
Left a few suggestions about comments, overall LGTM

File: exporter/datadogexporter/config/config.go
Comment (mx-psi) [reply exporter/datadogexporter/config/config.go]: > Btw, why do we set only_metadata as incompatible with host_metadata::hostname_source != first_resource?

I don't think this restriction makes a lot of sense, but I would rather think about it a bit more and change this in a different PR

**Review by KSerrania — APPROVED** [review]

**Comment by mx-psi** [comment]
This is ready to merge, but I am babysitting the CI until I finally get a test run without flakes

**Review by jpkrohling — COMMENTED** [review]
I'm approving on behalf of the other reviews, done by code owners.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
